core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -413,10 +413,8 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
     }
     if (protocols.contains(GroupType.STREAMS)) {
       if (processRoles.isEmpty || !isNewGroupCoordinatorEnabled) {
-        throw new ConfigException(s"The new '${GroupType.STREAMS}' rebalance protocol is only supported in KRaft cluster with the new group coordinator.")
+        warn(s"The new '${GroupType.STREAMS}' rebalance protocol is only supported in KRaft cluster with the new group coordinator.")
       }
-      warn(s"The new '${GroupType.STREAMS}' rebalance protocol is enabled along with the new group coordinator. " +
-        "This is part of the preview of KIP-1071 and MUST NOT be used in production.")
     }
```

Comment:
Why are you removing the warning that KIP-1071 must not be used in production?

Let's follow the same code as share groups above. I agree that we don't have to throw an exception here. We also shouldn't mention Kraft anymore, since ZooKeeper is gone.